src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -941,41 +932,25 @@ void AsyncRPCOperation_sendmany::sign_send_raw_transaction(UniValue obj)
         throw JSONRPCError(RPC_WALLET_ERROR, "Missing hex data for signed transaction");
     }
     std::string signedtxn = hexValue.get_str();
-
+    CDataStream stream(ParseHex(signedtxn), SER_NETWORK, PROTOCOL_VERSION);
+    CTransaction tx;
+    stream >> tx;
+    tx_ = tx;
     // Send the signed transaction
     if (!testmode) {
-        params.clear();
-        params.setArray();
-        params.push_back(signedtxn);
-        UniValue sendResultValue = sendrawtransaction(params, false);
-        if (sendResultValue.isNull()) {
-            throw JSONRPCError(RPC_WALLET_ERROR, "Send raw transaction did not return an error or a txid.");
-        }
-
-        std::string txid = sendResultValue.get_str();
-
+        CWalletTx wtx(pwalletMain, tx_);
+        pwalletMain->CommitTransaction(wtx, boost::none);
```

Comment:
Likewise here, use the `keyChange` from `add_taddr_change_output_to_tx()` if it is called.

Reply:
I've done most of the work for this locally, and it's possible that that I'm making this too complicated, so I wanted to touch base...

The `reserveKey` is defined as a `boost::optional<CReserveKey&>`, and because this is a reference, the following does not compile:
```
boost::optional<CReserveKey&> reserveKey = boost::none;
if (change > 0) {
    reserveKey = add_taddr_change_output_to_tx(change);
...
```
(assuming I change the definition of `add_taddr_change_output_to_tx` to return a `CReserveKey`)

I think that the solution is to change the `reserveKey` to a `boost::optional<CReserveKey>` in the places that we are using it, but I have a feeling that this will also require adding a copy constructor to `CReserveKey` to work properly.

Does this seem like the right approach?

Reply:
I looked into this today, and instead went with:
```cpp
CReserveKey keyChange(pwalletMain);
if (change > 0) {
    add_taddr_change_output_to_tx(keyChange, change);
...
```

This works because `CReserveKey` doesn't pull a key from the pool until `CReserveKey::GetReservedKey` is called in `add_taddr_change_output_to_tx()`, and `CommitTransaction` calls `CReserveKey::KeepKey` which does nothing if `CReserveKey::GetReservedKey` was not called.